gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -333,6 +362,21 @@ public Integer getMaxInboundMessageSize() {
       return maxInboundMessageSize;
     }
 
+    /** The maximum header list size allowed to be received on the channel. */
+    @BetaApi(
+        "The surface for maximum header list size is not stable yet and may change in the future.")
+    public Builder setMaxHeaderListSize(Integer max) {
```

Comment:
We should match the name that grpc will use for the final surface: `maxInboundMetadataSize` (+ `set`/`get`) (see https://github.com/grpc/grpc-java/issues/4050 )